scripts/release/github.ts

```diff
@@ -0,0 +1,337 @@
+import type { RestEndpointMethodTypes } from "@octokit/rest";
+import * as semver from "semver";
+
+import {
+  PR_FILES_STARTS_WITH,
+  NIGHTLY_BRANCH,
+  DEFAULT_BRANCH,
+} from "./constants";
+import { gql, graphqlWithAuth, octokit } from "./octokit";
+import type { MinimalTag } from "./utils";
+import { checkIfStringStartsWith, sortByDate } from "./utils";
+
+type PullRequest =
+  RestEndpointMethodTypes["pulls"]["list"]["response"]["data"][number];
+
+type PullRequestFiles =
+  RestEndpointMethodTypes["pulls"]["listFiles"]["response"]["data"];
+
+interface PrsMergedSinceLastTagOptions {
+  owner: string;
+  repo: string;
+  githubRef: string;
+}
+
+interface PrsMergedSinceLastTagResult {
+  merged: Awaited<ReturnType<typeof getPullRequestWithFiles>>;
+  previousTag: string;
+}
+
+export async function prsMergedSinceLastTag({
+  owner,
+  repo,
+  githubRef,
+}: PrsMergedSinceLastTagOptions): Promise<PrsMergedSinceLastTagResult> {
+  let tags = await getAllTags(owner, repo);
+  let { currentTag, previousTag } = getPreviousTagFromCurrentTag(
+    githubRef,
+    tags
+  );
+
+  let prs = await getMergedPRsBetweenTags(
+    owner,
+    repo,
+    previousTag,
+    currentTag,
+    currentTag.isPrerelease && previousTag.isPrerelease
+      ? NIGHTLY_BRANCH
+      : DEFAULT_BRANCH
```

Comment:
i think this is my main area of question, and i think i need an additional check for when going from a stable to nightly

nightly > nightly -> 'dev'
nightly > stable -> 'main'
stable -> nightly -> 'main' // but should be dev

Reply:
however this run makes me think otherwise 🙈 https://github.com/mcansh/nightly-release-test/runs/6937790319?check_suite_focus=true